File: VB/CustomizeReportDataExample.Module.Web/ReportWizardModifyController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.ReportsV2.Web
Imports CustomizeReportDataExample.Module.BusinessObjects

Namespace CustomizeReportDataExample.Module.Web
    Public Class ReportWizardModifyController
        Inherits ViewController

        Private reportServiceController As WebReportServiceController
        Public Sub New()
        End Sub
        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            reportServiceController = Frame.GetController(Of WebReportServiceController)()
            AddHandler reportServiceController.NewReportWizardShowing, AddressOf reportServiceController_NewReportWizardShowing
        End Sub
        Protected Overrides Sub OnDeactivated()
            RemoveHandler reportServiceController.NewReportWizardShowing, AddressOf reportServiceController_NewReportWizardShowing
            reportServiceController = Nothing
            MyBase.OnDeactivated()
        End Sub
        Private Sub reportServiceController_NewReportWizardShowing(ByVal sender As Object, ByVal e As WebNewReportWizardShowingEventArgs)
            If Not e.ReportDataType.Equals(GetType(MyReportDataV2)) Then
                Return
            End If
            Dim newReportParamsObject As New CustomReportWizardParameters(e.WizardParameters.Report, e.WizardParameters.ReportDataType)
            newReportParamsObject.Category = "Default"
            e.WizardParameters = newReportParamsObject
        End Sub
    End Class
End Namespace

File: CS/CustomizeReportDataExample.Module.Web/ReportWizardModifyController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.ReportsV2.Web;
using CustomizeReportDataExample.Module.BusinessObjects;

namespace CustomizeReportDataExample.Module.Web {
    public class ReportWizardModifyController : ViewController {
        WebReportServiceController reportServiceController;
        public ReportWizardModifyController() { }
        protected override void OnActivated() {
            base.OnActivated();
            reportServiceController = Frame.GetController<WebReportServiceController>();
            reportServiceController.NewReportWizardShowing +=
                reportServiceController_NewReportWizardShowing;
        }
        protected override void OnDeactivated() {
            reportServiceController.NewReportWizardShowing -=
                reportServiceController_NewReportWizardShowing;
            reportServiceController = null;
            base.OnDeactivated();
        }
        void reportServiceController_NewReportWizardShowing(object sender,
            WebNewReportWizardShowingEventArgs e) {
            if (!e.ReportDataType.Equals(typeof(MyReportDataV2))) return;
            CustomReportWizardParameters newReportParamsObject = new
                CustomReportWizardParameters(e.WizardParameters.Report, e.WizardParameters.ReportDataType);
            newReportParamsObject.Category = "Default";
            e.WizardParameters = newReportParamsObject;
        }
    }
}
